src/coreclr/vm/virtualcallstub.cpp

```diff
@@ -1817,11 +1815,18 @@ void VirtualCallStubManager::BackPatchWorkerStatic(PCODE returnAddress, TADDR si
     pMgr->BackPatchWorker(&callSite);
 }
 
-#if defined(TARGET_X86) && defined(TARGET_UNIX)
+#if defined(TARGET_X86)
 void BackPatchWorkerStaticStub(PCODE returnAddr, TADDR siteAddrForRegisterIndirect)
 {
     VirtualCallStubManager::BackPatchWorkerStatic(returnAddr, siteAddrForRegisterIndirect);
 }
+
+#ifdef CHAIN_LOOKUP
+ResolveCacheElem* __fastcall VSD_PromoteChainEntry(ResolveCacheElem *pElem)
+{
+    return VirtualCallStubManager::PromoteChainEntry(pElem);
+}
+#endif
```

Comment:
This add one level of indirection to get C-named method. We could use the manged name (`?PromoteChainEntry@VirtualCallStubManager@@CAPEAUResolveCacheElem@@PEAU2@@Z`) in the assembly instead if this is undesirable but I find it easier to read.

Pre-existing code is mixed in style. AMD64 code used mangled name for the `g_dispatch_cache_chain_success_counter` variable, ARM64 did not. Linux/x86 code used a forwarding function `BackPatchWorkerStaticStub` at one place but a mangled name for `VirtualCallStubManager::PromoteChainEntry`.

Reply:
If it's just dispatching the call to mangle-named function for build-time symbol resolution, then shouldn't the compiler optimize the pass-thru function in release mode?

Reply:
I suppose it will just tailcall. I doubt it would alias the symbols but it's possible.

Reply:
> Linux/x86 code used a forwarding function BackPatchWorkerStaticStub at one place but a mangled name for VirtualCallStubManager::PromoteChainEntry.

Can this be changed to follow uniform style accross the two methods and Win/Linux x86?

Reply:
Done